R/auxiliary.R

```diff
@@ -184,3 +184,243 @@ get_date_from_test_periods <- function(GeoLift, treatment_end_date, frequency =
+#' Synthetic Control Method. Augmentation through a prognostic function can improve
+#' fit and reduce L2 imbalance metrics.
+#' \itemize{
+#'          \item{"None":}{ ASCM is not augmented by a prognostic function. Defualt.}
+#'          \item{"Ridge":}{ Augments with a Ridge regression. Recommended to improve fit
+#'                           for smaller panels (less than 40 locations and 100 time-stamps.))}
+#'          \item{"GSYN":}{ Augments with a Generalized Synthetic Control Method. Recommended
+#'                          to improve fit for larger panels (more than 40 locations and 100
+#'                          time-stamps. }
+#'          \item{"best:}{ Fits the model with the lowest Scaled L2 Imbalance.}
+#'          }
+#' @param fixed_effects A logic flag indicating whether to include unit fixed
+#' effects in the model. Set to TRUE by default.
+#'
+#' @return
+#' Data frame with the locations and weights.
+#'
+#' @export
+GetWeights <- function(Y_id = "Y",
+                       time_id = "time",
+                       location_id = "location",
+                       X = c(),
+                       data,
+                       locations,
+                       pretreatment_end_time,
+                       model = "none",
+                       fixed_effects = TRUE) {
+
+  # Rename variables to standard names used by GeoLift
+  data <- data %>% dplyr::rename(
+    time = time_id,
+    Y = Y_id,
+    location = location_id
+  )
+
+  data$location <- tolower(data$location)
+  locations <- tolower(locations)
+
+  # Add filler rows if all time-stamps will be used for weight calculation
+  if(pretreatment_end_time == max(data$time)){
+    aux_rows <- data[data$time == max(data$time),]
+    aux_rows$time <- aux_rows$time + 1
+    data <- data %>% dplyr::add_row(aux_rows)
+    treatment_start_time <- max(data$time)
+    treatment_end_time <- max(data$time)
+  } else{
+    treatment_start_time <- pretreatment_end_time + 1
+    treatment_end_time <- max(data$time)
+  }
+
+  data_aux <- fn_treatment(data,
+                           locations = locations,
+                           treatment_start_time,
+                           treatment_end_time
+  )
+
+  if (length(X) == 0) {
+    fmla <- as.formula("Y ~ D")
+  } else if (length(X) > 0) {
+    fmla <- as.formula(paste(
+      "Y ~ D |",
+      sapply(list(X),
+             paste,
+             collapse = "+"
+      )
+    ))
+  }
+
+  # Single Augsynth
+  augsyn <- augsynth::augsynth(fmla,
+                               unit = location, time = time,
+                               data = data_aux,
+                               t_int = treatment_start_time,
+                               progfunc = model,
+                               scm = T,
+                               fixedeff = fixed_effects
+  )
+
+  results <- data.frame(
+    location = dimnames(augsyn$weights)[[1]],
+    weight = unname(augsyn$weights[, 1])
+  )
+
+  return(results)
```

Comment:
Artur, I understand what you are doing here.  However, isn't the use case for getting the weights of each location in the synth control when users actually run a GL?  If so, can't we just require a GeoLift object as an input, and output its weights for control locations?

If that's not the case, and imagine folks would like to know what are the weights per control locaation for each of the potential treatments (I don't know, may be overkill?), can't we leverage the output of each pvalueCalc ouptut (similar to the way in which we got the detected effect in zero) to get the weights?

When were you thinking that this function will be implemented?

Reply:
A common use-case is to obtain the weights before the treatment is applied which isn't possible at the moment. If you use all time-stamps of the data frame to calculate the weights you'll obtain an error with the dimensions of the augsynth input.